src/Memory/ILogStorage.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import * as CLM from '@conversationlearner/models'
+
+export interface LogQueryResult {
+    logDialogs: CLM.LogDialog[]
+    continuationToken: string | undefined
+}
+
+export interface ILogStorage {
+
+    /** Add a log dialog to storage */
+    Add(appId: string, logDialog: CLM.LogDialog): Promise<CLM.LogDialog>
+
+    /** Retrieve a log dialog from storage */
+    Get(appId: string, logDialogId: string): Promise<CLM.LogDialog | undefined>
+
+    /** Delete a log dialog in storage */
+    Delete(appId: string, lodDialogId: string): Promise<void>
+
+    NewSession(appId: string, logDialog: CLM.LogDialog): Promise<CLM.LogDialog>
```

Comment:
no need to have a "new session" here. ILogStorage shouldn't know anything about sessions.

Reply:
Yea, I don't have details but I thought this was supposed to be a generic storage interface that just happened to be storing log dialogs, but it could store anything.